sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -373,7 +374,11 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    * Create a [[DescribeQueryCommand]] logical command.
    */
   override def visitDescribeQuery(ctx: DescribeQueryContext): LogicalPlan = withOrigin(ctx) {
-    DescribeQueryCommand(visitQuery(ctx.query))
+    val query = ctx.query
+    val startIndex = query.start.getStartIndex
+    val endIndex = query.stop.getStopIndex
+    val interval = new Interval(startIndex, endIndex)
+    DescribeQueryCommand(query.start.getInputStream.getText(interval), visitQuery(ctx.query))
```

Comment:
@dilipbiswal . For consistency, could you improve `DescribeTableCommand` at least? For example, it will be useful for the following case.
```scala
spark-sql> EXPLAIN DESCRIBE TABLE t1;
== Physical Plan ==
*(1) Scan OneRowRelation[]

spark-sql> EXPLAIN DESCRIBE QUERY SELECT 1;
== Physical Plan ==
Execute DescribeQueryCommand
   +- DescribeQueryCommand SELECT 1
```

Reply:
@dongjoon-hyun Actually i had checked the explain output of `describe table`. I saw that `describe table` is special cased (isExplanableStatement) for explain and we purposely output as a one row relation. I wasn't sure of the history about why this is special cased and also how we should be the correct output. Thats why i only handled `describe query` in this pr.  If you can help with the expected output of `explain describe table` , i can try implementing it in this pr .. Please let me know.

Reply:
@dongjoon-hyun Can we take the DescribeTableCommand and DescribeColumnCommand as a follow-up ? I was thinking we could just capture the query text and show as the input to these commands like we are doing in this PR.

Reply:
Sure. No problem, @dilipbiswal . Please proceed as you want later.

Reply:
@dongjoon-hyun Thanks a lot. Actually after your comment yesterday, i had started to code it up.. Here is the link to the branch for your perusal. 
https://github.com/apache/spark/compare/master...dilipbiswal:describe_table_explain?expand=1
If you have some time and can quickly look through to see if the direction is okay ?